utilix/rundb.py

```diff
@@ -558,11 +560,21 @@ def _collection(experiment, collection, url=None, user=None, password=None, data
     if not database:
         database = uconfig.get('RunDB', f'{experiment}_database')
 
-    uri = f"mongodb://{user}:{password}@{url}"
-    c = pymongo.MongoClient(uri, readPreference='secondaryPreferred')
-    DB = c[database]
-    coll = DB[collection]
-    return coll
+    # build other client kwargs
+    max_pool_size = uconfig.get('RunDB', 'max_pool_size', fallback=100)
+    socket_timeout = uconfig.get('RunDB', 'socket_timeout', fallback=60000)
+    connect_timeout = uconfig.get('RunDB', 'connect_timeout', fallback=60000)
+
+    global MONGO_CLIENTS
```

Comment:
Declaring MONGO_CLIENTS as global is not needed, since you only change its contents and not the object the name is bound to in the global context.